### PR TITLE
feat: [PIE-5342]: Send parent entity details for template and pipeline APIs

### PR DESCRIPTION
**Pull request by depanshu-dhiman**
### Summary
Now applyTemplate, templateInputs, inputSets, variables, and execute-rerun call depends on a few extra params in case the parent entity is remote. This PR makes that change for all the references of such API calls.

Some APIs require these 5: 
  parentEntityConnectorRef?: string
  parentEntityRepoName?: string
  parentEntityAccountIdentifier?: string
  parentEntityOrgIdentifier?: string
  parentEntityProjectIdentifier?: string

Others only need 2:
  parentEntityConnectorRef?: string
  parentEntityRepoName?: string

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 4 Issues 
* Code issue 4 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11242)

**Review by yc147 — APPROVED** [review]

**Comment by depanshu-dhiman** [comment]
trigger cypress

File: src/modules/70-pipeline/components/PipelineStudio/TemplateBar/TemplateBar.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ export function TemplateBar(props: TemplateBarProps): JSX.Element {
       versionLabel: defaultTo(templateLinkConfig.versionLabel, ''),
       repoIdentifier,
       getDefaultFromOtherRepo: true,
-      ...createParentEntityQueryParams(storeMetadata, params, branch)
```
Comment (yc147) [inline src/modules/70-pipeline/components/PipelineStudio/TemplateBar/TemplateBar.tsx:75]: `TemplateBar` can be used `PipelineContext`. Pass `storeMetadata` in props.

**Review by yc147 — CHANGES_REQUESTED** [review]
TemplateBar can be used out side PipelineContext. Pass storeMetadata in props.

File: src/modules/72-templates-library/components/TemplateStudio/TemplateStudio.tsx [hunk 2]
```diff
@@ -306,7 +308,7 @@ export function TemplateStudio(): React.ReactElement {
   )
 
   return (
-    <TemplateVariablesContextProvider template={template}>
+    <TemplateVariablesContextProvider template={template} parentEntityData={parentEntityData}>
```
Comment (yc147) [inline src/modules/72-templates-library/components/TemplateStudio/TemplateStudio.tsx:311]: Pass the storeMetaData to `TemplateVariablesContextProvider`

**Comment by depanshu-dhiman** [comment]
retrigger jest

**Review by yc147 — DISMISSED** [review]

**Review by YounixIO — DISMISSED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by yc147 — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

**Pull request merged**